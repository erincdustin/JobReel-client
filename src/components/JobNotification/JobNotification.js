import React from 'react';
import Button from '../../components/Button/Button';
import jobReelApiService from '../../services/jobreel-api-service';
import jobReelContext from '../../context/JobReelContext';

class JobNotification extends React.Component {

  static contextType = jobReelContext

  state = {
    job_title: this.props.title, 
    company: this.props.company, 
    city: this.props.city,  
    state: this.props.state,  
    url: this.props.url,  
    status: this.props.status,  
    description: this.props.desc, 
    job_id: this.props.job_id, 
    date_added: this.props.date_added, 
    date_applied: this.props.date_applied, 
    user_id: this.props.user_id, 
    notification: this.props.notification
  }

  handleHideJob() {
    const { job_title, company, city, state, url, status, description, date_applied, job_id, date_added, user_id } = this.state
    let editedJob = {
        job_title,
        company, 
        city, 
        state, 
        url, 
        status, 
        description,
        job_id,
        date_added,
        date_applied,
        user_id,
        notification: false
    }
    jobReelApiService.editJob(editedJob, job_id)
    this.context.updateJob(editedJob)
    this.setState({ notification: false })
  }

  renderNotification(){
    let today = new Date();
    let applied = new Date(this.state.date_applied)
    let diff = Math.abs((today - applied) / (1000 * 60 * 60 * 24))
    let rounded = Math.round(diff)
    if (this.state.date_applied && (this.state.notification === true) && (rounded > 7)) {
    return <p key={this.state.job_id}>You applied to {this.state.company} {rounded} days ago.  Have you heard back? <button onClick={() => this.handleHideJob()}>Clear</button></p>
      }
  }
  
  render() {
    return (
      <div className="job-notification">
        {this.renderNotification()}
      </div>
    )
  }
}

export default JobNotification;