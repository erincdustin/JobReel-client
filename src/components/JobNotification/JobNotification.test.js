import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import JobNotification from './JobNotification'

const props = {
  job_id: 2,
  user_id: 1,
  job_title: 'UI Designer',
  company: 'Company B',
  city: 'Austin',
  state: 'TX',
  date_added: '2019-08-14T17:18:19.306Z',
  url: 'http://www.google.com',
  description: 'UI job',
  status: 'Interested',
  date_applied: '2019-08-16T17:18:19.306Z',
  notification: true
  }

describe('JobNotification smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<JobNotification {...props}/>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('JobNotification snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<JobNotification {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})