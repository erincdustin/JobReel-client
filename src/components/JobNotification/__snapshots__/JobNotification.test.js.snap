// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobNotification snapshot test renders the UI as expected 1`] = `
<div
  className="job-notification"
>
  <p
    key="2"
  >
    You applied to 
    Company B
     
    13
     days ago.  Have you heard back? 
    <button
      onClick={[Function]}
    >
      Clear
    </button>
  </p>
</div>
`;
